### ISSUE TITLE
Automate example notebook generation

### ISSUE THREAD

**Issue by w-k-jones** [tobac-project/tobac#232]
We should set up an action to automate the generation of the example notebooks with releases, to avoid issues such as that raised in #231 

Guide from developer meeting on how to automate running jupyter notebooks: https://innerjoin.bit.io/automate-jupyter-notebooks-on-github-9d988ecf96a6

**Comment by lettlini** [comment]
I think this is a great idea. Besides avoiding issues like #231, I think it would also greatly improve the workflow when working with Jupyter notebooks in combination with Git. 
Removing all output from a notebook before committing is quite simple and can be done by something like `jupyter nbconvert --clear-output`.

**Comment by freemansw1** [comment]
I think this is a great idea, and maybe resolves some of the original headaches that led to the split of the tutorials repo from the main repo.

**Comment by lettlini** [comment]
Lately I have been working on the notebooks a lot so I would like to share some thoughts:
- having an action that handles the generation of output on merges to the main branch automatically would be brilliant
- I think it would be equally as important to switch to notebooks without any output in the development branches such that diff & merge becomes possible / easier
- the implementation of these changes probably needs to be well coordinated and definitely should happen after any imminent releases

I can offer to look into creating a draft for a github action. However, I have no experience with this and will probably need a little help.

**Comment by lettlini** [comment]
I am not sure whether this is necessary but perhaps we should also think about setting up a CI workflow that runs on development branches and makes sure that notebooks run without any errors. In essence this would be the same workflow as for the main branch except that it does not commit anything. 

This would make sure that if there are any errors in the notebook that would prevent the automatic notebook generation from running successfully we would be made aware of that before trying to merge into the main branch.

**Comment by w-k-jones** [comment]
Great! Just had a look, they both look good. My only thought is to change the triggering events that work then see if the workflows run as expected. My thoughts for the triggering workflows are:

check_notebooks.yml:
`on: [push, pull_request]`
so that it runs for any push/PR opened

generate_notebooks.yml:
```
on:
  pull_request:
    types:
      - closed
```
so that it runs when a PR is merged

is there any other times when these workflows could be run?

**Comment by w-k-jones** [comment]
Good call, I had neglected to consider that it would also run on a closed PR that isn't merged. That looks good, perhaps only running generate_notebooks.yml on the main branch might also be a good idea to limit overuse. We could also include the `on: workflow_dispatch` for triggering manually for testing/regenerating at will

**Comment by freemansw1** [comment]
I've moved this to the 1.5 milestone and will close once #258 is merged
